customers/models.py

```diff
@@ -35,6 +37,174 @@ def get_queryset(self):
+                    },
+                    "comment": "Laskunumero: 247509 RAMSAYRANTA A 004"
+                }
+            ],
+            "organization": {  <-- either a dict like here or it will not be in this customer's dict at all
+                "type": "company",  <-- values from OrganizationType enum
+                "name": "Nice Profitable Firm Ltd.",
+                "address": "Mannerheimintie 1 A 1",
+                "postal_code": "00100",
+                "city": "Helsinki"
+            },
+        }
+
+        And returns dict where key is the customer_id and value is the UUID of created profile object
+        """
+        from leases.models import BerthLease
+        from payments.models import Order
+        from resources.models import Berth, Harbor
+
+        result = {}
+        for index, customer_data in enumerate(data):
+            try:
+                if not customer_data.get("id"):
+                    raise Exception(_("No customer ID provided"))
+
+                customer = self.create(
+                    id=customer_data.get("id"), comment=customer_data.get("comment"),
+                )
+                customer.refresh_from_db()
+
+                if organization := customer_data.get("organization"):
+                    organization_type = OrganizationType(organization.get("type"))
+                    Organization.objects.create(
+                        customer=customer,
+                        organization_type=organization_type,
+                        name=organization.get("name"),
+                        address=organization.get("address"),
+                        postal_code=organization.get("postal_code"),
+                        city=organization.get("city"),
+                        business_id="-"
+                        if organization_type == OrganizationType.COMPANY
+                        else "",
+                    )
+
+                boats = []
+                for boat in customer_data.get("boats", []):
+                    boat_type = BoatType.objects.language("fi").get(
+                        translations__name=boat.get("boat_type")
+                    )
+                    boats.append(
+                        Boat.objects.create(
+                            owner=customer,
+                            name=boat.get("name"),
+                            boat_type=boat_type,
+                            registration_number=boat.get("registration_number"),
+                            width=Decimal(boat.get("width")),
+                            length=Decimal(boat.get("length")),
+                            draught=Decimal(boat.get("draught"))
+                            if boat.get("draught")
+                            else None,
+                            weight=Decimal(boat.get("weight"))
+                            if boat.get("weight")
+                            else None,
+                        )
+                    )
+
+                leases = []
```

Comment:
why do you store leases in this list? I see that for `boats` you need it when checking for a specific boat from the lease, but this `lease` list does not seem to be used anywhere

Reply:
Good point 👍